### PR TITLE
MISUV-1438: call using nino to get business details

### PR DESCRIPTION
**Pull request by scullythornton1**
…detail to retrieve mtditid

### CODE REVIEW

File: app/models/core/MtdbsaModel.scala [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+case class MtdbsaModel(mtdbsa: Option[String]) extends MtdbsaResponse
+
+case class MtdbsaErrorModel(status: Int, reason: String) extends MtdbsaResponse
+
+object MtdbsaModel {
+  implicit val format: Format[MtdbsaModel] = Json.format[MtdbsaModel]
+}
+
+object MtdbsaErrorModel {
+  implicit val format: Format[MtdbsaErrorModel] = Json.format[MtdbsaErrorModel]
```
Comment (AlexRimmerHMRC) [inline app/models/core/MtdbsaModel.scala:33]: Can you remove this as it's not being used

File: app/models/incomeSourceDetails/IncomeSourceDetailsResponseModel.scala [hunk 2]
```diff
@@ -30,6 +31,7 @@ case class IncomeSourceDetailsError(status: Int, reason: String) extends IncomeS
 object IncomeSourceDetailsModel {
 
   def applyWithFields(nino: String,
+                      mtdbsa: Option[String],
```
Comment (AlexRimmerHMRC) [inline app/models/incomeSourceDetails/IncomeSourceDetailsResponseModel.scala:34]: according to the api spec, mtdbsa is not an optional field

File: app/services/GetBusinessDetailsService.scala [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+@Singleton
+class GetBusinessDetailsService @Inject()(val getBusinessDetailsConnector: GetBusinessDetailsConnector) {
+
+  def getBusinessDetails(nino: String)(implicit headerCarrier: HeaderCarrier): Future[IncomeSourceDetailsResponseModel] = {
+    Logger.debug("[getBusinessDetailsService][getBusinessDetails] - Requesting Income Source Details from Connector")
+    getBusinessDetailsConnector.getBusinessDetails(nino).map {
+      case success: IncomeSourceDetailsModel =>
+        Logger.debug(s"[getBusinessDetailsService][getBusinessDetails] - Retrieved Get Business Details:\n\n$success")
+        Logger.debug(s"[getBusinessDetailsService][getBusinessDetails] - Converting to IncomeSourceDetails Model")
```
Comment (AlexRimmerHMRC) [inline app/services/GetBusinessDetailsService.scala:36]: can you remove this second debug log as it's not providing us with any extra information

File: app/models/core/MtdbsaModel.scala (hunk 1 above)
Comment (scullythornton1) [reply app/models/core/MtdbsaModel.scala:33]: done

File: app/services/GetBusinessDetailsService.scala (hunk 3 above)
Comment (scullythornton1) [reply app/services/GetBusinessDetailsService.scala:36]: done

File: app/models/incomeSourceDetails/IncomeSourceDetailsResponseModel.scala (hunk 2 above)
Comment (scullythornton1) [reply app/models/incomeSourceDetails/IncomeSourceDetailsResponseModel.scala]: done fixing

**Pull request merged**